lib/rules/valid-local-dependency.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * @fileoverview Checks existence of local dependencies in the package.json
+ * @author Kendall Gassner
+ */
+'use strict';
+const path = require('path');
+const {
+    isPackageJson,
+    extractPackageObjectFromAST
+} = require('../processors/PackageJsonProcessor');
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description:
+                'Checks existence of local dependencies in the package.json',
+            category: 'Best Practices',
+            recommended: true
+        },
+        fixable: null // or "code" or "whitespace"
+    },
+
+    create: function(context) {
+        return {
+            'Program:exit': node => {
+                if (!isPackageJson(context.getFilename())) {
+                    return;
+                }
+                const sourceCode = context.getSourceCode();
+                const packageRoot = extractPackageObjectFromAST(node);
+                const original = JSON.parse(sourceCode.getText(packageRoot));
+                const {
+                    dependencies,
+                    peerDependencies,
+                    devDependencies
+                } = original;
+
+                const depObjs = [
+                    Object.entries(dependencies || {}),
+                    Object.entries(peerDependencies || {}),
+                    Object.entries(devDependencies || {})
+                ];
+
+                depObjs.map(obj =>
+                    obj.map(([key, value]) => {
+                        const response = localPath => {
+                            const filePath = path.join(
+                                context
+                                    .getFilename()
+                                    .replace(/package\.json/g, '/'),
+                                value.replace(new RegExp(localPath, 'g'), ''),
+                                '/package.json'
+                            );
+
+                            try {
+                                if (!require.resolve(filePath)) {
+                                    context.report({
+                                        node: packageRoot,
+                                        message: `The package ${key} does not exist given the specified path: ${value}.`
+                                    });
+                                }
+                            } catch (e) {
+                                context.report({
+                                    node: packageRoot,
+                                    message: `The package ${key} does not exist given the specified path: ${value}.`
```

Comment:
[Thoughts] It'd be nice if it used a different, more explicit message if the only difference is casing. But that's probably better as a follow-up issue. I bet it's surprisingly hard to get right 😄 .

Reply:
_Technically_ it will catch spelling mistakes and non-existing packages.... but package managers should catches those so the rule is mostly useful for casing errors.

Reply:
Good point. I think I'd personally still interpret it for both, since it gives the nice errors before package managers would.